catroid/src/main/java/org/catrobat/catroid/utils/IconsUtil.java

```diff
@@ -292,48 +293,24 @@ public static void addIcons(ViewGroup viewGroup, String category) {
 		}
 	}
 
-	public static void setMainMenuIconSize(Context context, View contentView) {
+	public static void setMenuIconSize(View contentView, Rect desiredIconSize) {
 		if (!isLargeSize()) {
 			return;
 		}
 
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_continue,
-				R.drawable.ic_main_menu_continue);
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_new,
-				R.drawable.ic_main_menu_new);
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_programs,
-				R.drawable.ic_main_menu_programs);
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_help,
-				R.drawable.ic_main_menu_help);
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_web,
-				R.drawable.ic_main_menu_community);
-		setLeftDrawableSize(context, contentView, largeIconSizeMainMenu, R.id.main_menu_button_upload,
-				R.drawable.ic_main_menu_upload);
-	}
-
-	public static void setProgramMenuIconSize(Context context, View contentView) {
-		if (!isLargeSize()) {
-			return;
+		ViewGroup content = (ViewGroup) contentView;
+		for (int i = 0; i < content.getChildCount(); i++) {
+			View child = content.getChildAt(i);
+
+			if (child instanceof LinearLayout || child instanceof RelativeLayout || child instanceof ScrollView) {
```

Comment:
Wouldn't it be better to check for instanceof ViewGroup here? Since LinearLayout, RelativeLayout etc. are all ViewGroups and I think they should all be called recursively? E.g. RecyclerView would be missing in this check.. but with ViewGroup we wouldn't have to worry about all other container views.

Reply:
You are absolutely right. Fixed.